app/models/template_input.rb

```diff
@@ -71,7 +71,7 @@ def prevent_delete_if_template_is_locked
   end
 
   def check_if_template_is_locked
-    if template&.locked
+    if template&.run_template_changes_validation?
```

Comment:
I've not realized the method is private on `Template`

Reply:
made it public